app/main/helpers/validation_tools.py

```diff
@@ -33,66 +42,77 @@ def question_errors(self, question_id):
                 return
 
         for rule in question_content["validations"]:
-            if not self.test(question, rule["name"]):
+            if not self.test(question_id, question, rule["name"]):
                 return rule["message"]
 
-        # TODO We can't update file URLs until we have the write API,
-        # so uploaded file must have the same extension as the
-        # current one for now
-        current_extension = get_extension(self.service.get(question_id, ''))
-        if get_extension(question.filename) != current_extension:
-            return u"Uploaded file format should be: %s" % current_extension
-
-    def test(self, question, rule):
+    def test(self, question_id, question, rule):
         if not hasattr(self, rule):
             raise ValueError("Validation rule " + rule + " not found")
-        return getattr(self, rule)(question)
+        return getattr(self, rule)(question_id, question)
 
-    def answer_required(self, question):
+    def answer_required(self, question_id, question):
         not_empty = len(question.read(1)) > 0
         question.seek(0)
         return not_empty
 
-    def file_can_be_saved(self, question):
-        file_name = self._generate_file_name(question)
+    def file_can_be_saved(self, question_id, question):
+        file_path = generate_file_name(
+            self.service['supplierId'],
+            self.service['id'],
+            question_id,
+            question.filename
+        )
+
+        if self.service[question_id]:
+            existing_path = urlparse.urlsplit(
+                self.service[question_id]
+            ).path.lstrip('/')
+        else:
+            existing_path = None
 
         self.uploader.save(
-            'documents/{}'.format(
-                self.service['supplierId'],
-            ),
-            file_name,
-            question
+            file_path,
+            question,
+            existing_path
         )
 
+        full_url = urlparse.urljoin(
+            main.config['DOCUMENTS_URL'],
+            file_path
+        )
+
+        self.clean_data[question_id] = full_url
+
         return True
```

Comment:
Is it possible to return `false` here if `self.uploader.save` fails for any reason? Would be better to show the user an error rather than an exception.